bids/layout/writing.py

```diff
@@ -6,57 +6,14 @@
 import os
 import re
 import sys
+from string import Formatter
+from itertools import product
 from ..utils import splitext, listify
 from os.path import join, dirname, exists, islink, isabs, isdir
 
-__all__ = ['replace_entities', 'build_path', 'write_contents_to_file']
+__all__ = ['build_path', 'write_contents_to_file']
 
-
-def replace_entities(entities, pattern):
```

Comment:
Are we confident nobody will ever want to use this directly, outside of the context of path-building? I don't remember the original motivation for exposing this at module level, but I suspect that was the consideration....

Reply:
Having this method separated included an extra run of the regex per entity. Now everything is done in the internal loop, and the actual replacement uses python's formatting (which is accessible to anyone). I would see some interest in getting the pattern once interpreted instead and allow the user do the formatting... but honestly I'm not sure it is worth the separation.

Reply:
Cool, let's leave it as-is for now then.